src/Aspire.Hosting.Azure.Sql/AzureSqlServerResource.cs

```diff
@@ -108,11 +115,76 @@ public override ProvisionableResource AddAsExistingResource(AzureResourceInfrast
     public override void AddRoleAssignments(IAddRoleAssignmentsContext roleAssignmentContext)
     {
         var infra = roleAssignmentContext.Infrastructure;
-        var postgres = (SqlServer)AddAsExistingResource(infra);
+        var sqlserver = (SqlServer)AddAsExistingResource(infra);
 
-        var principalId = roleAssignmentContext.PrincipalId;
+        // The name of the user assigned identity for the container app
         var principalName = roleAssignmentContext.PrincipalName;
+        var principalId = roleAssignmentContext.PrincipalId;
 
-        AzureSqlExtensions.AddActiveDirectoryAdministrator(infra, postgres, principalId, principalName);
+        var sqlServerAdmin = UserAssignedIdentity.FromExisting("sqlServerAdmin");
+        sqlServerAdmin.Name = AdminName.AsProvisioningParameter(infra);
+        infra.Add(sqlServerAdmin);
+
+        foreach (var database in Databases.Keys)
+        {
+            // Create a unique resource identifier for the script resource
+            var hash = new XxHash3();
+            hash.Append(Encoding.UTF8.GetBytes(database));
+            hash.Append(Encoding.UTF8.GetBytes(infra.AspireResource.GetBicepIdentifier()));
+
+            var scriptResource = new SqlServerScriptProvisioningResource($"script_{Convert.ToHexString(hash.GetCurrentHash()).ToLowerInvariant()}");
```

Comment:
Shouldn't we do this in bicep itself? Use `BicepFunction.GetUniqueString()` (https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-string#uniquestring)

Reply:
OH this is the bicep identifier.

Reply:
Dont you want to call Intrastructure.NormalizeIndentifier here?

Reply:
Made the name and identifier nicer:

The identifier is descriptive and unique in the module.
The name is technical, short and unique.

```
resource script_mysqlserver_todosdb 'Microsoft.Resources/deploymentScripts@2023-08-01' = {
  name: take('script-${uniqueString('mysqlserver', 'todosdb', resourceGroup().id)}', 24)
```